compiler/resolution/functionResolution.cpp

```diff
@@ -7601,6 +7639,17 @@ static void instantiate_default_constructor(FnSymbol* fn) {
         }
       }
     }
+
+    if (AggregateType* ct = toAggregateType(fn->_this->type)) {
+      if (ct->instantiationStyle == DEFINES_INITIALIZER &&
```

Comment:
I hope that calling the default constructor from the user-defined one is temporary... it is, right?